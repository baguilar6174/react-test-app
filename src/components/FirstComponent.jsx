import PropTypes from 'prop-types';
import React from 'react'

const FirstComponent = ({title, subtitle, name}) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <p>{subtitle}</p>
      <p>{name}</p>
    </>
  )
}

FirstComponent.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
}

FirstComponent.defaultProps = {
  name: "Bryan",
  // title: "No title",
  subtitle: "No subtitle",
}

export default FirstComponent;