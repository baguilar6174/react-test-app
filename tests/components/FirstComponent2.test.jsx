import { render, screen } from "@testing-library/react";
import FirstComponent from "../../src/components/FirstComponent";

describe('Test on <FirstComponent/>', () => {

  const title = "Hi, I am Bryan";
  const subtitle = "subtitle";

  test('should be match with the snapshot', () => {
    const { container } = render(
      <FirstComponent title={title} />
    );
    expect(container).toMatchSnapshot();
  });

  test('should show "Hi, I am Bryan"', () => {
    render(<FirstComponent title={title} />);
    // screen.debug
    expect(screen.getByText(title)).toBeTruthy();
  });

  test('should show title in <h1></h1>', () => {
    render(<FirstComponent title={title} />);
    // screen.debug
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(title);
  });

  test('should show the subtitle sent in props', () => {
    render(<FirstComponent title={title} subtitle={subtitle} />);
    expect(screen.getAllByText(subtitle)).toBeTruthy();
  });

});