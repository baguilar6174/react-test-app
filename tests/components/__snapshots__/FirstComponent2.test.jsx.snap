// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <FirstComponent/> should be match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hi, I am Bryan
  </h1>
  <p>
    No subtitle
  </p>
  <p>
    Bryan
  </p>
</div>
`;
